packages/spear-cli/src/plugins/spear-i18n.ts

```diff
@@ -0,0 +1,191 @@
+import { parse } from "node-html-parser"
+import { bufferDeepCopy, loadFile } from "../util.js"
+import { AssetFile, Component, Element, HookApi, SpearSettings, SpearState } from "../interfaces/HookCallback"
+
+
+interface I18nSettings {
+    default?: string,
+}
+type I18nWords = Map<string, string>
+type I18nLanguages = Map<string, I18nWords>
+
+const i18nSettings: I18nSettings = {
+    default: "",
+}
+const i18nLanguages: I18nLanguages = new Map<string, I18nWords>()
+
+// This plugin will replace i18n syntax.
+// SettingsFile:
+//   JavaScript Object files which has key and value like following.
+//   export default {
+//     settings: {
+//       default: "jp",
+//       fallback: "jp",
+//     },
+//     "jp": [
+//       { title: "ブログです" },
+//       { description: "ブログサイトです" },
+//     ],
+//     "en": [
+//       { title: "Blog Site" },
+//       { description: "This is blog site." },
+//     ]
+//   }
+//  Or Yaml object file which has key and value like the bellow:
+//    settings:
+//      default: "jp"
+//      fallback: "jp"
+//    lang:
+//      jp:
+//        - title: ブログだよ
+//        - description: ブログサイトです
+//      en:
+//        - title: Blog
+//        - description: This is blog site.
+
+export function spearI18n(settingsFile?: string): HookApi {
+    // Use configuration and afterBuild hook for generating SEO.
+    return {
+        // Build internal variable for converting i18n
+        configuration: async function(settings: SpearSettings) {
+            if (settingsFile !== "") {
+                try {
+                    const settingFileContent = await loadFile(`${settings.rootDir}/${settingsFile}`)
+                    if (settingFileContent) {
+                        Object.keys(settingFileContent).forEach(key => {
+                            if (key.toLowerCase() === "settings") {
+                                i18nSettings.default = settingFileContent[key].default
+                            } else if (key === "lang") {
+                                // Collect the languages definition.
+                                const languagesObject = settingFileContent[key]
+                                Object.keys(languagesObject).forEach(lang => {
+                                    const words = languagesObject[lang]
+                                    const convWords: I18nWords = new Map<string, string>()
+                                    words.forEach(w=> {
+                                        Object.keys(w).forEach(subKey => {
+                                            convWords.set(subKey, w[subKey])
+                                        })
+                                    })
+                                    i18nLanguages.set(lang.toLowerCase(), convWords)
+                                })
+                            }

```

Comment:
How about throw Error at this case?
(or waring)

```suggestion
                            } else {
                                throw new Error(`invalid key name. [${key}]`)
                            }

```

Reply:
throw error could be annoying, maybe warning would be better

Reply:
@qst-exe @yoannes 

Thank you~!!!
Both of your guys suggestion is true.

This i18n implementation is plugins. So If plugin throws error(or Runtime error), Spear should catch this error and display it as warning then Spear generating process should be continue.

So I added `try ~ catch` in the Spear: https://github.com/unimal-jp/spear/pull/110/files#diff-d4f1cb69151af8368d22300c3e4f25858a8ba0ad8a502a04b09ceb3a490e918bR309-R317